src/averageAges.js

```diff
@@ -36,8 +45,23 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChildren = false) {
+  if (withChildren === false) {
+    const womenArray = people.filter(woman => woman.sex === 'f');
+    const yearsSum = womenArray.reduce((years, woman) =>
+      years + (woman.died - woman.born), 0);
+
+    return yearsSum / womenArray.length;
```

Comment:
This logic is repeated in every function. Let's put this in a separate function `calculateAverageAge` that will take an array of numbers and a total number